web/src/message_list_view.js

```diff
@@ -130,6 +140,19 @@ function update_group_date(group, message_container, prev) {
     group.date_unchanged = same_day(message_container?.msg, prev?.msg);
 }
 
+function update_message_year(curr_message, prev_message) {
+    if (prev_message === undefined) {
+        curr_message.display_year = false;
+        return;
+    }
+
+    if (same_year(curr_message, prev_message)) {
+        curr_message.display_year = false;
+    } else {
+        curr_message.display_year = true;
+    }
```

Comment:
Is this just `curr_message.display_year = !same_year(curr_message, prev_message)`?